File: mainFolder/Adventurer.java
public class Adventurer {
    private String name;
    private int health;
    private int attackPower;

    public Adventurer(String name, int health, int attackPower) {
        this.name = name;
        this.health = health;
        this.attackPower = attackPower; //code to create new adventurer
    }    

    public String getName() {
        return name; //gets name
    }

    public int getHealth() {
        return health; //gets health
    }

    public int getAttackPower() {
        return attackPower; //gets attackpower 
    }

    public void setHealth(int health) {
        this.health = health; //sets health
    }

    public void attack(Adventurer opponent) {
        int damage = this.attackPower; //logic for attacking an opponent
        opponent.setHealth(opponent.getHealth() - damage);
        System.out.println(this.name + " attacks " + opponent.getName());
    }
}

File: mainFolder/adventurer.rs
#[derive(Clone)]
#[derive(PartialEq)]
pub struct Adventurer {
    name: String,
    health: i32,
    attack_power: i32,
}

impl Adventurer {
    pub fn new(name: String, health: i32, attack_power: i32) -> Self {
        Adventurer {
            name,
            health,
            attack_power,
        }
    }

    pub fn get_name(&self) -> &str {
        &self.name
    }

    pub fn get_health(&self) -> i32 {
        self.health
    }

    pub fn get_attack_power(&self) -> i32 {
        self.attack_power
    }

    pub fn set_health(&mut self, health: i32) {
        self.health = health;
    }

    pub fn attack(&self, opponent: &mut Adventurer) {
        let damage = self.attack_power;
        opponent.set_health(opponent.get_health() - damage);
        println!("{} attacks {}!", self.name, opponent.get_name());
    }
}